### DISCUSSION TITLE
How can I get the build number as a environment variable? [Flutter] [iOS]

### DISCUSSION THREAD

**Discussion by DanielJaramillo94** [codemagic-ci-cd/codemagic-docs#1369] [Q&A] [answered]
If I do not set the flag `--build-number` when running `flutter build ipa --release`, then the **CFBundleVersion** is `build_number` I set on the  `pubspec.yaml` of my project (which format is `{major}.{minor}.{patch}+{build_number}`, as you well know), that is an integer.

I want to access that `build_number` in CodeMagic so I can, for example, set `flutter build ipa --release --build-number=$BUILD_NUMBER.1`.

How can I do that?

**Comment by himesh-cm** [comment]
Hi @DanielJaramillo94, Yes, you can set. It will automatically fetch 

```
flutter build ipa --release --build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=$PROJECT_BUILD_NUMBER
```
You can read more about build versioning [here](https://docs.codemagic.io/knowledge-codemagic/build-versioning/).

**Reply by DanielJaramillo94** [reply]
Hi @himesh-cm, thank you for your response.

`$PROJECT_BUILD_NUMBER` is a built-in environment variable and stores the number of builds we have made on CodeMagic. This variable does not correspond to what I am asking.

**Comment by DanielJaramillo94** [comment, marked as answer]
I managed to do it using custom environment variables.
I use this pre-build script to store the `build_number` in a variable called `FLUTTER_BUILD_NUMBER`.

`
version=$(grep -E "^version: .+\+.+" /Users/builder/clone/pubspec.yaml);
buildVersion=$(echo $version | awk -F "+" '{print $2}');
echo "FLUTTER_BUILD_NUMBER=$buildVersion" >> $CM_ENV;
`

More info about custom environment variables [here](https://docs.codemagic.io/yaml-basic-configuration/using-environment-variables/).
